util/reggen/countermeasure.py

```diff
@@ -0,0 +1,71 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Dict, Sequence
+
+from .lib import check_keys, check_str, check_list
+
+# TO BE DEFINED
+CM_ASSETS = [
+    'FOO'
+]
+
+# TO BE DEFINED
+CM_TYPES = [
+    'BAR'
+]
+
+
+class CounterMeasure:
+    def __init__(self, name: str, desc: str):
+        self.name = name
+        self.desc = desc
+
+    @staticmethod
+    def from_raw(what: str, raw: object, bname: str) -> 'CounterMeasure':
+        rd = check_keys(raw, what, ['name', 'desc'], [])
+
+        name = check_str(rd['name'], 'name field of ' + what)
+        desc = check_str(rd['desc'], 'desc field of ' + what)
+
+        name_split = name.split('.')
+
+        if len(name_split) != 3:
+            raise ValueError('The countermeasure ID {} ({}) has the '
+                             'wrong format.'
+                             .format(name, what))
+
+        if bname.upper() != name_split[0]:
+            raise ValueError('The block name of countermeasure {} ({})'
+                             'is {} but should be {}.'
+                             .format(name, what, name_split[0], bname))
+
+        if name_split[1] not in CM_ASSETS:
+            raise ValueError('The asset identifier {} of countermeasure '
+                             '{} ({}) is not defined.'
+                             .format(name_split[1], name, what))
+
+        if name_split[2] not in CM_TYPES:
+            raise ValueError('The type identifier {} of countermeasure '
+                             '{} ({}) is not defined.'
+                             .format(name_split[2], name, what))
+
+        return CounterMeasure(name, desc)
+
+    @staticmethod
+    def from_raw_list(what: str,
+                      raw: object,
+                      bname: str) -> Sequence['CounterMeasure']:
```

Comment:
Not that it matters, but did `List['CounterMeasure']` not work here?

Reply:
This works as well (I've used Sequence since most other objects use this in ip_block). I'll change it.

Reply:
Ah, that's probably because they are returning objects of different subclasses. In mypy, `Sequence` is covariant (and immutable) and `List` is not. See https://mypy.readthedocs.io/en/stable/generics.html#variance-of-generics for the details.

Reply:
Ah gotcha, thanks for the pointer.